components/color-picker/index.jsx

```diff
@@ -0,0 +1,582 @@
+	 * * `relative` - No styling or portals will be used. Menus will be positioned relative to their triggers. This is a great choice for HTML snapshot testing.
+	 */
+	menuPosition: PropTypes.oneOf([
+		'absolute',
+		'overflowBoundaryElement',
+		'relative',
+	]),
+	/**
+	 * An array of hex color values which is used to set the options of the
+	 * swatch tab of the colorpicker popover.
+	 * To specify transparent, use empty string as a value.
+	 */
+	swatchColors: PropTypes.arrayOf(PropTypes.string),
+	/**
+	 * Determines which tab is visible when dialog opens. Use this prop with `base` variant only.
+	 * Defaults to `swatch` tab.
+	 */
+	selectedTab: PropTypes.oneOf('swatches, custom'),
+	/**
+	 * Selects which tabs are present for the colorpicker.
+	 * * `base`: both swatches and custom tabs are present
+	 * * `swatches`: only swatch tab is present
+	 * * `custom`: only custom tab is present
+	 * _Tested with snapshot testing._
+	 */
+	variant: PropTypes.oneOf(['base', 'swatches', 'custom']),
+	/**
+	 * Current color in hexadecimal string, including # sign (eg: "#000000")
+	 */
+	value: PropTypes.string,
+	/**
+	 * Current working color in hexadecimal string, including # sign (eg: "#000000")
+	 */
+	valueWorking: PropTypes.string,
+};
+
+const defaultProps = {
+	assistiveText: {
+		saturationValueGrid:
+			'Use arrow keys to select a saturation and brightness, on an x and y axis.',
+		hueSlider: 'Select Hue',
+	},
+	events: {},
+	labels: {
+		blueAbbreviated: 'B',
+		cancelButton: 'Cancel',
+		customTab: 'Custom',
+		customTabActiveWorkingColorSwatch: 'Working Color',
+		customTabTransparentSwatch: 'Transparent',
+		greenAbbreviated: 'G',
+		hexLabel: 'Hex',
+		invalidColor: 'The color entered is invalid',
+		invalidComponent: 'The value needs to be an integer from 0-255',
+		redAbbreviated: 'R',
+		submitButton: 'Done',
+		swatchTab: 'Default',
+	},
+	menuPosition: 'absolute',
+	swatchColors: [
+		'#e3abec',
+		'#c2dbf7',
+		'#9fd6ff',
+		'#9de7da',
+		'#9df0c0',
+		'#fff099',
+		'#fed49a',
+		'#d073e0',
+		'#86baf3',
+		'#5ebbff',
+		'#44d8be',
+		'#3be282',
+		'#ffe654',
+		'#ffb758',
+		'#bd35bd',
+		'#5779c1',
+		'#5679c0',
+		'#00aea9',
+		'#3cba4c',
+		'#f5bc25',
+		'#f99221',
+		'#580d8c',
+		'#001970',
+		'#0a2399',
+		'#0b7477',
+		'#0b6b50',
+		'#b67e11',
+		'#b85d0d',
+	],
+	selectedTab: 'swatches',
+	variant: 'base',
+};
+
+class ColorPicker extends React.Component {
+	static displayName = COLOR_PICKER;
+	static propTypes = propTypes;
+	static defaultProps = defaultProps;
+
+	constructor(props) {
+		super(props);
+
+		this.generatedId = this.props.id || shortid.generate();
+		const workingColor = ColorUtils.getNewColor(
+			{
+				hex:
+					this.props.valueWorking ||
+					this.props.value ||
+					this.props.swatchColors[0],
+			},
+			this.props.events.onValidateWorkingColor
+		);
+		this.state = {
+			currentColor: this.props.value || this.props.swatchColors[0],
```

Comment:
Note here - if no value passed in, we'll use the first color in the set for the current color, instead of defaulting to transparent.

Reply:
It just felt strange to set the current color to "transparent" by default - which is not even in the set of pre-defined colors. By selecting the first one, when the popover opens, the top right color is focused, which feels right as a starting point. I also assume that most consumer won't be selecting transparent as their main color but that's just an assumption.